src/c_api.cpp

```diff
@@ -1035,6 +1035,33 @@ int LGBM_DatasetCreateFromMat(const void* data,
 }
 
 
+static inline std::vector<int32_t> CreateSampleIndices(const Config& config, int32_t total_nrow) {
+  Random rand(config.data_random_seed);
+  int sample_cnt = static_cast<int>(total_nrow < config.bin_construct_sample_cnt ? total_nrow : config.bin_construct_sample_cnt);
+  return rand.Sample(total_nrow, sample_cnt);
+}
+
+
+int LGBM_SampleIndices(int32_t total_nrow,
+                const char* parameters,
+                void* out) {
```

Comment:
```suggestion
int LGBM_SampleIndices(int32_t total_nrow,
                       const char* parameters,
                       void* out) {
```

Please align these arguments vertically, similar to the style used in the rest of this file. This makes it a bit easier to read through the list of parameters.